sql-plugin/src/main/scala/org/apache/spark/sql/rapids/complexTypeExtractors.scala

```diff
@@ -87,15 +87,15 @@ class GpuGetArrayItemMeta(
   override def convertToGpu(
       arr: Expression,
       ordinal: Expression): GpuExpression =
-    GpuGetArrayItem(arr, ordinal)
+    GpuGetArrayItem(arr, ordinal, SQLConf.get.ansiEnabled)
```

Comment:
This is wrong. In version prior to 3.1.1 the default value should be false, not based off of the `ansiEnabled` config. Otherwise we will fail when spark does not.

Reply:
Set to false and add comment.